lib/apiservers/engine/backends/volume.go

```diff
@@ -58,7 +103,68 @@ func (v *Volume) VolumeRm(name string) error {
 		if _, ok := err.(*storage.RemoveVolumeConflict); ok {
 			return derr.NewRequestConflictError(fmt.Errorf("Volume is in use"))
 		}
-		return derr.NewErrorWithStatusCode(fmt.Errorf("Server error form portlayer"), http.StatusInternalServerError)
+		return derr.NewErrorWithStatusCode(fmt.Errorf("Server error from portlayer: %s", err), http.StatusInternalServerError)
 	}
 	return nil
 }
+
+//Utility Functions
+
+func fillDockerVolumeModel(volume *models.VolumeResponse, labels map[string]string) *types.Volume {
+	dockerVol := types.Volume{
+		Driver:     volume.Driver,
+		Name:       volume.Name,
+		Labels:     labels,
+		Mountpoint: volume.SelfLink,
+	}
+	return &dockerVol
+}
+
+func validateDriverArgs(args map[string]string, model *models.VolumeRequest) error {
+	//volumestore name validation
+	storeName, ok := args[optsVolumeStoreKey]
+	if !ok {
+		storeName = "default"
+	}
+	model.Store = storeName
+
+	//capacity validation
+	capstr, _ := args[optsCapacityKey]
```

Comment:
looks like I need to make my unit test better. It should have caught something like this.